MISRA.md

```diff
@@ -0,0 +1,26 @@
+# MISRA Compliance
+
+The MQTT library files conform to the [MISRA C:2012](https://www.misra.org.uk/MISRAHome/MISRAC2012/tabid/196/Default.aspx)
+guidelines, with some noted exceptions. Compliance is checked with Coverity static analysis.
+Deviations from the MISRA standard are listed below:
+
+### Ignored by Coverity Configuration
+| Deviation | Category | Justification |
+| :-: | :-: | :-: |
+| Directive 4.5 | Advisory | Allow names that MISRA considers ambiguous (such as LogInfo and LogError) |
+| Directive 4.8 | Advisory | Allow inclusion of unused types. Header files for a specific port, which are needed by all files, may define types that are not used by a specific file. |
+| Directive 4.9 | Advisory | Allow inclusion of function like macros. The `assert` macro is used throughout the library for parameter validation, and logging is done using function like macros. |
+| Rule 2.4 | Advisory | Allow unused tags. Some compilers warn if types are not tagged. |
+| Rule 2.5 | Advisory | Allow unused macros. Library headers may define macros intended for the application's use, but are not used by a specific file. |
+| Rule 3.1 | Required | Allow nested comments. C++ style `//` comments are used in example code within Doxygen documentation blocks. |
```

Comment:
I would prefer not to disable rule 3.1 in the global configuration since that will disable detection of other violations.  How many occurrences of doxygen sample code exist which contain comments?  Could those be suppressed with inline annotations?

Reply:
There are almost 200 occurrences of `//` within doxygen sample code, and no they cannot be suppressed with inline annotations.

Reply:
We have _many_ occurrences of doxygen-based sample code comments in the library, basically for each public API function in the [core_mqtt_serializer.h](https://github.com/FreeRTOS/coreMQTT/blob/master/source/include/core_mqtt_serializer.h) and [core_mqtt.h](https://github.com/FreeRTOS/coreMQTT/blob/master/source/include/core_mqtt.h) files